feature_engine/selection/target_mean_encoding.py

```diff
@@ -0,0 +1,400 @@
+from sklearn.base import TransformerMixin, BaseEstimator
+from sklearn.utils.validation import check_is_fitted
+from sklearn.model_selection import train_test_split
+from sklearn.metrics import roc_auc_score, r2_score
+
+from feature_engine.variable_manipulation import _find_all_variables, _define_variables
+from feature_engine.dataframe_checks import _is_dataframe
+
+import pandas as pd 
+
+class TargetMeanEncoding(BaseEstimator, TransformerMixin):
+    """
```

Comment:
we need to think of a different name for this transformer. This transformer aims to select features, so it need to me more or less clear from the name, what the functionallity does.

Something like TargetMeanEncoderFeatureSelector(), or maybe you are more creative?

Reply:
Your suggested name sounds good to me. Thanks.